cavern/src/main/java/org/opencadc/cavern/nodes/CephFSQuotaPlugin.java

```diff
@@ -0,0 +1,155 @@
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *
+ ************************************************************************
+ */
+
+package org.opencadc.cavern.nodes;
+
+import org.apache.log4j.Logger;
+import org.opencadc.util.fs.ExtendedFileAttributes;
+import org.opencadc.util.fs.PathUtil;
+
+import java.io.IOException;
+import java.nio.file.Path;
+
+
+/**
+ * QuotaPlugin implementation that can get and set specific values for interacting with CephFS.  This class relies
+ * on the Extended Attributes to be set on the appropriate folder.
+ */
+public class CephFSQuotaPlugin implements QuotaPlugin {
+    private static final Logger LOGGER = Logger.getLogger(CephFSQuotaPlugin.class);
+    private static final String NAMESPACE = "ceph";
+    private static final String QUOTA_ATTR_KEY = "quota.max_bytes";
+    private static final String RECURSIVE_SIZE_ATTR_KEY = "dir.rbytes";
+
+
+    @Override
+    public Long getBytesUsed(Path directory) {
+        LOGGER.debug("getBytesUsed: " + directory);
+        if (directory == null) {
+            throw new IllegalArgumentException("Cannot get size of null directory");
+        }
+
+        try {
+            final String recursiveBytesUsedValue =
+                    ExtendedFileAttributes.getFileAttribute(directory, CephFSQuotaPlugin.RECURSIVE_SIZE_ATTR_KEY,
+                                                            CephFSQuotaPlugin.NAMESPACE);
+            final long recursiveBytesUsed = Long.parseLong(recursiveBytesUsedValue);
+            LOGGER.debug("getBytesUsed: " + directory + ": OK");
+            return recursiveBytesUsed;
+        } catch (IOException ioException) {
+            LOGGER.warn("Unable to find recursive folder size of " + directory + " ("
+                        + CephFSQuotaPlugin.NAMESPACE + "." + CephFSQuotaPlugin.QUOTA_ATTR_KEY + ")");
+            return null;
+        }
+    }
+
+    /**
+     * Obtain the quota amount, in bytes, set at the highest level of the provided path.  This method will traverse,
+     * beginning at the first Path element, until it finds a directory containing the CephFS Quota attribute.
+     * Returning a null value means no quota is set, and users can write until the underlying storage is full.
+     *
+     * @param directory directory to check
+     * @return null if the directory input is null, or if no directory contains the quota attribute.
+     */
+    @Override
+    public Long getQuota(Path directory) {
+        LOGGER.debug("getQuota: " + directory);
+        if (directory == null) {
+            throw new IllegalArgumentException("Cannot check quota of null directory");
+        }
+
+        try {
+            final Long quotaInBytes = PathUtil.scanQuotaFromPath(directory, 0, CephFSQuotaPlugin.QUOTA_ATTR_KEY,
```

Comment:
This is wrong. the quota is a property of a specific node and this should only check the argument Path.

Most calls will return null except when `directory` has the ceph quota prop

Reply:
Understood.  I had thought the patch checking would translate into this property checking as well.  Done.